File: 20190730_Tuesday/20190730.rb
p "alias x='ruby 20190730_Tuesday/20190730.rb'" 

# Ruby Regex#scan returns a flat array. ^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$ => [["...","...","...","...","..."]];
require "minitest/autorun"
require_relative "lrc2subrip.rb"
require 'ostruct'

describe "Pawn Race" do
    it "Pass the tests" do
        [
            OpenStruct.new({
                lrcLyrics: ["[00:12.00] Happy birthday dear coder,", 
                        "[00:17.20] Happy birthday to you!"],
                songLength: "00:00:20",
                answer: ["1", 
                    "00:00:12,000 --> 00:00:17,200", 
                    "Happy birthday dear coder,", 
                    "", 
                    "2", 
                    "00:00:17,200 --> 00:00:20,000", 
                    "Happy birthday to you!"]
            }),
        ].each{|x| assert_equal x.answer, Task.new.lrc2subRip(x.lrcLyrics, x.songLength)}
    end
end

File: 20190730_Tuesday/20190730.js
console.log("alias x='node 20190730_Tuesday/20190730.js'");

// JS String.prototype.match and RegExp.prototype.exec and RegExp.prototype.test. (JS uses undefined when Ruby uses nil)

console.log("[00:12.00] Happy birthday dear coder,".match(/^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g));
console.log(
    /^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g.exec("[00:12.00] Happy birthday dear coder,")
    );
    console.log(
        /^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g.test("[00:12.00] Happy birthday dear coder,")
        );


        console.log("[00:12.00]".match(/^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g));
console.log(
    /^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g.exec("[00:12.00]")
    );
    console.log(
        /^\[(\d+):(\d+)\.(\d+)\]( (.*)){0,1}$/g.test("[00:12.00]")
        );